packages/tpetra/core/src/Tpetra_Details_PackTraits.hpp

```diff
@@ -165,17 +167,26 @@ struct PackTraits {
   /// \param numEnt [in] Number of entries to pack.
   ///
   /// \return The number of bytes used to pack \c inBuf into \c outBuf.
+  KOKKOS_INLINE_FUNCTION
   static size_t
   packArray (const output_buffer_type& outBuf,
              const input_array_type& inBuf,
-             const size_t numEnt)
+             const size_t numEnt,
+             int& stat,
+             std::unique_ptr<std::string>& errStr)
   {
-#ifdef HAVE_TPETRA_DEBUG
-    TEUCHOS_TEST_FOR_EXCEPTION(
-      static_cast<size_t> (inBuf.dimension_0 ()) < numEnt,
-      std::invalid_argument, "PackTraits::packArray: inBuf.dimension_0() = "
-      << inBuf.dimension_0 () << " < numEnt = " << numEnt << ".");
-#endif // HAVE_TPETRA_DEBUG
+    stat = 0;
+    if (static_cast<size_t> (inBuf.dimension_0 ()) < numEnt) {
+      std::ostringstream os;
```

Comment:
See above.  For functions intended to run possibly on the GPU, you should only collect an error code at this point.  Don't turn it into a string until you get back to host.